presto-hive/src/main/java/com/facebook/presto/hive/s3select/S3SelectJsonRecordReader.java

```diff
@@ -63,4 +65,10 @@ protected OutputSerialization buildOutputSerialization()
 
         return selectObjectOutputSerialization;
     }
+
+    @Override
+    protected boolean shouldEnableScanRange()
+    {
+        return CompressionType.NONE.equals(getCompressionType());
```

Comment:
We can actually assert the same invariant here actually. If `CompressionType != NONE` then we _must_ be scanning the entire file, otherwise there's a bug and we should fail instead of producing incorrect query results. Basically, this check should make no behavioral decisions about scan range dynamically because it's already too late after `InternalHiveSplitFactory` has decided a file (or all files in the partition/table) is splittable and generated more than one split for it. All we can do at this point is detect a bug and fail.

Reply:
Same as a above, leftover, this check is covered by the check from `InternalHiveSplitFactory`.